lib/samples/apply_function_to_raster_from_file/apply_function_to_raster_from_file.dart

```diff
@@ -0,0 +1,114 @@
+  final _mapViewController = ArcGISMapView.createController();
+  // Raster layer to display raster data on the map.
+  late RasterLayer _rasterLayer;
+  // A flag for when the map view is ready and controls can be used.
+  var _ready = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: Stack(
+        children: [
+          ArcGISMapView(
+            controllerProvider: () => _mapViewController,
+            onMapViewReady: onMapViewReady,
+          ),
+          // Display a progress indicator and prevent interaction until state is ready.
+          LoadingIndicator(visible: !_ready),
+        ],
+      ),
+    );
+  }
+
+  Future<void> onMapViewReady() async {
+    // Create a map with a standard imagery basemap style.
+    final map = ArcGISMap.withBasemapStyle(BasemapStyle.arcGISImageryStandard);
+    // Add map to the map view.
+    _mapViewController.arcGISMap = map;
+    // Load the raster layer.
+    await loadRasterLayer();
+    // Add the raster layer to the map.
+    map.operationalLayers.add(_rasterLayer);
+    // Set the viewpoint to the center of the raster layer's full extent.
+    final fullExtent = _rasterLayer.fullExtent;
+    if (fullExtent != null) {
+      final center = fullExtent.center;
+      const scale = 80000.0;
+      await _mapViewController.setViewpointCenter(center, scale: scale);
+    }
+    // Set the ready state variable to true to enable the sample UI.
+    setState(() => _ready = true);
+  }
+
+  Future<void> loadRasterLayer() async {
+    // Download the raster file.
+    await downloadSampleData(['b051f5c3e01048f3bf11c59b41507896']);
+    // Get the application documents directory.
+    final appDir = await getApplicationDocumentsDirectory();
+    // Create a Raster from the local tif file.
+    final shastaElevationRaster = Raster.withFileUri(
+      Uri.file('${appDir.absolute.path}/Shasta_Elevation/Shasta_Elevation.tif'),
+    );
+    // Load the raster.
+    await shastaElevationRaster.load();
+    // Load JSON file from assets.
+    final rasterColorJson = await rootBundle.loadString(
+      'assets/color_raster_function.json',
+    );
+    // Create a RasterFunction.
+    final rasterFunction = RasterFunction.fromJson(rasterColorJson);
+    if (rasterFunction != null) {
+      final arguments = rasterFunction.arguments;
+      if (arguments != null) {
+        final rasterNames = arguments.rasterNames;
+        // Set the raster function arguments as required by the function used.
+        arguments.setRaster(
+          name: rasterNames[0],
+          raster: shastaElevationRaster,
+        );
+        arguments.setRaster(
+          name: rasterNames[1],
+          raster: shastaElevationRaster,
+        );
+        // Create a Raster from the raster function.
+        final raster = Raster.withFunction(rasterFunction);
+        // Load the Raster Layer.
+        _rasterLayer = RasterLayer.withRaster(raster);
```

Comment:
I'm comparing the Qt version of this sample. They set opacity to .5 on the _rasterLayer. We want to do the same thing, so that it blends the images together. (This will mean the screenshot has to be updated as well.)  https://github.com/Esri/arcgis-maps-sdk-samples-qt/blob/f4f5d85c8583a83e2067969dc39e1b255c2e0be9/CppSamples/Layers/RasterFunctionFile/RasterFunctionFile.cpp#L91